pandas/core/dtypes/missing.py

```diff
@@ -51,6 +54,9 @@
 )
 from pandas.core.dtypes.inference import is_list_like
 
+if TYPE_CHECKING:
+    from pandas.core.arrays.base import ExtensionArraySupportsAnyAll
```

Comment:
ExtensionArraySupportsAnyAll is a terrible hack to avoid Protocol. (at a time when we supported Python 3.7)

Probably best to avoid using this for now as the more we use it the greater chance we will run into issues with false positives)

Reply:
Removed it but added an ignore comment.